frontend/src/concepts/pipelines/apiHooks/mlmd/usePipelineRunMlmdContext.ts

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { Context } from '~/third_party/mlmd';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import useFetchState, {
+  FetchState,
+  FetchStateCallbackPromise,
+  NotReadyError,
+} from '~/utilities/useFetchState';
+import { GetContextByTypeAndNameRequest } from '~/third_party/mlmd/generated/ml_metadata/proto/metadata_store_service_pb';
+import { FAST_POLL_INTERVAL } from '~/utilities/const';
+
+const KFP_V2_RUN_CONTEXT_TYPE = 'system.PipelineRun';
+
+const useMlmdContext = (
```

Comment:
Ideally, this is unit tested.

Reply:
We need to mock up some MLMD objects to unit test this, I prefer to push it later to add with other MLMD hooks.

Reply:
I'm not sure why it would be done with other MLMD hooks. Why do you prefer that?

Reply:
I am just unsure how to write unit test for this... any ideas?

Reply:
OK, I tried to create some unit tests but it doesn't work good and I don't know why. I decided to create a JIRA ticket for it and just add some comments on this hook for better future use so that I can wrap this PR up.

Reply:
https://issues.redhat.com/browse/RHOAIENG-5510